bukkit/src/main/java/net/vyhub/tasks/TRewards.java

```diff
@@ -0,0 +1,182 @@
+package net.vyhub.tasks;
+
+import net.vyhub.VyHubPlatform;
+import net.vyhub.abstractClasses.ARewards;
+import net.vyhub.abstractClasses.AUser;
+import net.vyhub.entity.AppliedReward;
+import net.vyhub.entity.Reward;
+import net.vyhub.event.VyHubPlayerInitializedEvent;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.event.player.PlayerRespawnEvent;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.logging.Level.INFO;
+import static java.util.logging.Level.WARNING;
+
+public class TRewards extends ARewards implements Listener {
+
+    public TRewards(VyHubPlatform platform, AUser aUser) {
+        super(platform, aUser);
+    }
+
+    @EventHandler
+    public void onPlayerInit(VyHubPlayerInitializedEvent event) {
+        Player player = event.getPlayer();
+        getPlatform().executeAsync(() -> {
+            getPlayerReward(player);
+
+            List<String> eventList = new LinkedList<>();
+            eventList.add("CONNECT");
+            eventList.add("SPAWN");
+
+            getPlatform().executeBlocking(() -> {
+                executeReward(eventList, player.getUniqueId().toString());
+            });
+        });
+    }
+
+    @Override
+    public void getPlayerReward(Object object) {
+
+    }
+
+    public synchronized void executeReward(List<String> events, String playerID) {
+        if (getRewards() == null) {
+            return;
+        }
+
```

Comment:
The logic of executing rewards should be in common.
Only things like executing a command etc. should be in the specific plugin